catalog/ftp/src/main/java/ddf/catalog/ftp/ftplets/FtpRequestHandler.java

```diff
@@ -151,65 +232,118 @@ public FtpletResult onUploadStart(FtpSession session, FtpRequest request)
         }
 
         session.write(new DefaultFtpReply(FtpReply.REPLY_150_FILE_STATUS_OKAY,
-                STOR_REQUEST + " " + fileName));
+                requestTypeString + " " + fileName));
         LOGGER.debug("Replying to client with code 150 - file status okay");
 
-        try (FileBackedOutputStream outputStream = new FileBackedOutputStream(1000000);
-                final AutoCloseable ac = outputStream::reset) {
-            DataConnection dataConnection = connFactory.openConnection();
-            dataConnection.transferFromClient(session, outputStream);
+        if (isDotFile(request.getArgument())) {
+            DataConnection dataConnection;
+            try {
+                dataConnection = connFactory.openConnection();
+            } catch (Exception e) {
+                LOGGER.debug("Error getting the output data stream from the FTP session", e);
+                throw new IOException("Error getting the output stream from FTP session");
```

Comment:
@glenhein Sorry, I didn't notice the Exception vs IOException there. It probably makes more sense to catch IOException here and then rethrow that, and below catch generic "Exception" and then wrap that in an FtpException. 

In fact i think if you catch "Exception" before FtpException it will never hit the FtpException case as FtpException subclasses Exception.